src/x86/hob.c

```diff
@@ -0,0 +1,241 @@
+        case EfiUnusableMemory:
+            wolfBoot_printf("EfiUnusableMemory\r\n");
+            break;
+        case EfiACPIReclaimMemory:
+            wolfBoot_printf("EfiACPIReclaimMemory\r\n");
+            break;
+        case EfiACPIMemoryNVS:
+            wolfBoot_printf("EfiACPIMemoryNVS\r\n");
+            break;
+        case EfiMemoryMappedIO:
+            wolfBoot_printf("EfiMemoryMappedIO\r\n");
+            break;
+        case EfiMemoryMappedIOPortSpace:
+            wolfBoot_printf("EfiMemoryMappedIOPortSpace\r\n");
+            break;
+        case EfiPalCode:
+            wolfBoot_printf("EfiPalCode\r\n");
+            break;
+        case EfiPersistentMemory:
+            wolfBoot_printf("EfiPersistentMemory\r\n");
+            break;
+        case EfiMaxMemoryType:
+            wolfBoot_printf("EfiMaxMemoryType\r\n");
+            break;
+        default:
+            wolfBoot_printf("Unknown efi_memory_type\r\n");
+            break;
+    }
+}
+
+void dump_efi_hob(const struct efi_hob *hob) {
+    if (hob == NULL) {
+        wolfBoot_printf("Invalid HOB pointer\r\n");
+        return;
+    }
+
+    wolfBoot_printf("HOB Type: %d\r\n", (uint32_t)hob->u.header.hob_type);
+    wolfBoot_printf("HOB Length: %u\r\n", (uint32_t)hob->u.header.hob_length);
+
+    switch (hob->u.header.hob_type) {
+    case EFI_HOB_TYPE_HANDOFF:
+        wolfBoot_printf("HOB Type: EFI_HOB_TYPE_HANDOFF\r\n");
+        // Handle specific fields for this HOB type
+        break;
+    case EFI_HOB_TYPE_MEMORY_ALLOCATION:
+        wolfBoot_printf("HOB Type: EFI_HOB_TYPE_MEMORY_ALLOCATION\r\n");
+        dump_guid(&hob->u.memory_allocation.alloc_descriptor.name);
+        dump_efi_memory_type(
+            hob->u.memory_allocation.alloc_descriptor.memory_type);
+        wolfBoot_printf(
+            "Memory Base Address: 0x%x\r\n",
+            (uint32_t)
+            hob->u.memory_allocation.alloc_descriptor.memory_base_address);
+        wolfBoot_printf(
+            "Memory Length: %u\r\n",
+            (uint32_t)hob->u.memory_allocation.alloc_descriptor.memory_length);
+        // Handle specific fields for this HOB type
+        break;
+    case EFI_HOB_TYPE_RESOURCE_DESCRIPTOR:
+        wolfBoot_printf("HOB Type: EFI_HOB_TYPE_RESOURCE_DESCRIPTOR\r\n");
+        wolfBoot_printf("Resource Type: 0x%x\r\n",
+                        hob->u.resource_descriptor.resource_type);
+        dump_guid(&hob->u.resource_descriptor.owner);
+        wolfBoot_printf("Physical Start: 0x%x_%x\r\n",
+                        (uint32_t)(hob->u.resource_descriptor.physical_start >> 32),
+                        (uint32_t)hob->u.resource_descriptor.physical_start);
+        wolfBoot_printf("Resource Length: 0x%x_%x\r\n",
+                        (uint32_t)(hob->u.resource_descriptor.resource_length >> 32),
+                        (uint32_t)hob->u.resource_descriptor.resource_length
+                        );
+        // Handle specific fields for this HOB type
```

Comment:
Use C style `/* */` comments.

Reply:
the code was unused, it has been removed